### ISSUE TITLE
Interpolation extensions; how to?

### ISSUE THREAD

**Issue by Bertg** [ruby-i18n/i18n#157]
Hi all,

I've been working quite some time with Tr8n now; and we feel it is no longer suiting our needs (mostly regarding implementation). Lots of its features we like. Some of the features we'd like to bring over to i18n, as we think it makes sense in this project.

The first batch we'd like to bring over, is basically the ability to do this:

``` ruby
# :key = "There %{count| is, are} %{count|| ninja, ninjas} in this room"
I18n.t(:key, count: 1) # => "There is 1 ninja in this room"

# :key = "If you want the ninjas to go, [link: click here]"
I18n.t(:key, link: ->(text) { link_to(text, remove_ninja_path) } ) # => "If you want the ninjas to go, <a href="/ninja/remove">click here</a>"
```

There are even more interpolation extensions in Tr8n, but I'm not sure If all of these would fit within I18n. 

Basically most of the code to do this already exists, and it's now just a way of figuring out how to bring it over to the i18n project. 
- Is there interest in this type of functionality?
- Should it be part of the I18n gem?
- If not part of the I18n gem, what would be the best way to package this type of functionality?

Let me know what you think!
